pkg/pool-weighted/contracts/WeightedPoolProtocolFees.sol

```diff
@@ -105,6 +105,10 @@ abstract contract WeightedPoolProtocolFees is BaseWeightedPool, ProtocolFeeCache
             );
     }
 
+    function getATHRateProduct() external view returns (uint256) {
```

Comment:
Well, now `getLastInvariant` should also be external (so we can call it directly from the test, vs exposing it in a Mock).

Can we move put them together near the top (showing the external interface first), and add @notice (since they're external)

Reply:
We weren't exposing it through the mock but I've changed `getLastInvariant` to external from public